### PR TITLE
drivers: adc: Update drivers to use devicetree Kconfig symbol

### PR DESCRIPTION
**Pull request by galak**
Update adc drivers to use DT_HAS_<compat>_ENABLED Kconfig symbol
to expose the driver and enable it by default based on devicetree.

We remove 'depend on' Kconfig for symbols that would be implied by
the devicetree node existing.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

File: drivers/adc/Kconfig.mcux [hunk 1]
```diff
@@ -5,36 +5,34 @@
...
+	depends on DT_HAS_NXP_LPC_LPADC_ENABLED
 	help
 	  Enable the MCUX LPADC driver.
 
-config ADC_MCUX_ETC
-	bool "MCUX ADC ETC driver"
-	depends on HAS_MCUX_ADC_ETC
-	help
-	  Enable the MCUX ADC ETC driver.
-
```
Comment (henrikbrixandersen) [inline drivers/adc/Kconfig.mcux:37]: Are you sure this can be removed? It was added in c343da51612dbdc40902a29e5898846eb91effce.

Comment (henrikbrixandersen) [reply drivers/adc/Kconfig.mcux:37]: cc @danieldegrasse @mmahadevan108

**Review by henrikbrixandersen — DISMISSED** [review]

File: drivers/adc/Kconfig.mcux (hunk 1 above)
Comment (bbilas) [reply drivers/adc/Kconfig.mcux]: This symbol doesn't have any driver and currently it's only used to enable `adc_etc` driver from the nxp_hal manually so it must stay as it is.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by bbilas — APPROVED** [review]

File: drivers/adc/Kconfig.test [hunk 2]
```diff
@@ -2,9 +2,8 @@
...
 # SPDX-License-Identifier: Apache-2.0
 
-DT_COMPAT_VND_ADC := vnd,adc
-
 # Hidden option for turning on the dummy driver for vnd,adc devices
 # used in testing.
 config ADC_TEST
-	def_bool $(dt_compat_enabled,$(DT_COMPAT_VND_ADC))
+	def_bool DT_HAS_VND_ADC_ENABLED
+	depends on DT_HAS_VND_ADC_ENABLED
```
Comment (nandojve) [inline drivers/adc/Kconfig.test:9]: Why is this different from others?

**Review by nandojve — CHANGES_REQUESTED** [review]

File: drivers/adc/Kconfig.test (hunk 2 above)
Comment (nandojve) [reply drivers/adc/Kconfig.test:9]: found answer in another PR review.

**Review by nandojve — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**